src/LinuxTracing/ThreadStateManagerTest.cpp

```diff
@@ -19,17 +20,22 @@ using orbit_grpc_protos::ThreadStateSlice;
 
 TEST(ThreadStateManager, OneThread) {
   constexpr pid_t kTid = 42;
+  constexpr pid_t kWasBlockedByTid = 420;
+  constexpr pid_t kWasBlockedByPid = 4200;
   ThreadStateManager manager;
   std::optional<ThreadStateSlice> slice;
 
-  manager.OnInitialState(100, kTid, ThreadStateSlice::kRunnable);
+  manager.OnInitialState(100, kTid, ThreadStateSlice::kRunnable, kWasBlockedByTid,
```

Comment:
I'm not familiar with this code. Why OnInitialState has BlockerTid and Pid. Isn't that the initial state?

Reply:
`OnInitialState` can create a state that is interruptable sleep, runnable, or running. So, in some cases you might need to add a blocker tid and pid. It's not the initial state, it allows you to create any state you want.

Reply:
I would have that values constructed by default inside OpenState instead. And I wouldn't include these values in OnInitialState neither in this test.